reahl-component/reahl/component_dev/test_dbutils.py

```diff
@@ -57,20 +59,88 @@ def test_finding_database_control(dbcontrol_fixture):
         SystemControl(fixture.config)
 
 
-@with_fixtures(DBControlFixture)
-def test_database_control_settings(dbcontrol_fixture):
+class URIScenarios(Fixture):
+
+    @scenario
+    def all_settings_given(self):
+        self.uri = 'myprefix://theuser:thepasswd@thehost:123/thedb'
+        self.database_name = 'thedb'
+        self.user_name = 'theuser'
+        self.password = 'thepasswd'
+        self.host = 'thehost'
+        self.port = 123
+
+    @scenario
+    def not_all_settings_given(self):
+        self.uri = 'myprefix://theuser@thehost/thedb'
+        self.database_name = 'thedb'
+        self.user_name = 'theuser'
+        self.password = None
+        self.host = 'thehost'
+        self.port = None
+
+    @scenario
+    def minimal(self):
+        self.uri = 'myprefix:///:memory:'
+        self.database_name = ':memory:'
+        self.user_name = None
+        self.password = None
+        self.host = None
+        self.port = None
+
+
+@with_fixtures(DBControlFixture, URIScenarios)
+def test_database_control_settings(dbcontrol_fixture, uri_fixture):
```

Comment:
Just for consistency....perhaps call uri_fixture uri_scenario - that's what we do elsewhere.